config/initializers/000-development_reload_warnings.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Development helper which prints a warning when you edit a non-autoloaded ruby file.
+# These include initializers, middleware, plugin.rb files, and more.
+# Launch the server with AUTO_RESTART=1 to automate restarts.
+if Rails.env.development? && !Rails.configuration.cache_classes
+  paths = [
+    *Dir["#{Rails.root}/app/*"].reject { |path| path.end_with? "/assets" },
+    "#{Rails.root}/config",
+    "#{Rails.root}/lib",
+    "#{Rails.root}/plugins"
+  ]
+
+  auto_restart = ENV["AUTO_RESTART"]
```

Comment:
Minor but this can be moved into the conditional which is the only place this is used.